src/engine.cpp

```diff
@@ -81,7 +81,7 @@ Engine::Engine(VehicleType type, EngineID base)
 		/* 'power' defaults to zero, so we also have to default to 'wagon' */
 		if (type == VEH_TRAIN) this->u.rail.railveh_type = RAILVEH_WAGON;
 		/* Set model life to maximum to make wagons available */
-		this->info.base_life = 0xFF;
+		this->info.base_life = CalendarTime::MAX_YEAR;
```

Comment:
MAX_YEAR isn't 0xFF. There are some tests at lines 659 and 673 that test for 0xFF. And the values from NewGRF are uint8_t, so there 0xFF has special meaning too.

Reply:
Then I'll leave them as `0xFF` for now.